viziphant/unitary_event_analysis.py

```diff
@@ -0,0 +1,404 @@
+        if n < n_neurons - 1:
+            axis3.axhline((trial + 2) * (n + 1), lw=params_dict['lw'],
+                          color='k')
+    axis3.set_xlim(xlim_left, xlim_right)
+    axis3.set_ylim(0, (trial + 2) * (n + 1) + 1)
+    axis3.xaxis.set_major_locator(MaxNLocator(integer=True))
+    axis3.set_yticks(y_ticks_list)
+    axis3.set_yticklabels(y_ticks_labels_list, fontsize=params_dict['fsize'])
+    axis3.set_ylabel('Trial', fontsize=params_dict['fsize'])
+
+    print('plotting Coincidence Rates ..')
+    axis4 = plt.subplot(6, 1, 4, sharex=axis1)
+    axis4.set_title('Coincidence Rates')
+    empirical_coincidence_rate = joint_surprise_dict['n_emp'] / \
+                                 (window_size.rescale('s').magnitude * n_trials)
+    expected_coincidence_rate = joint_surprise_dict['n_exp'] / \
+                                (window_size.rescale('s').magnitude * n_trials)
+    axis4.plot(center_of_analysis_window, empirical_coincidence_rate,
+               label='empirical', lw=params_dict['lw'], color='c')
+    axis4.plot(center_of_analysis_window, expected_coincidence_rate,
+               label='expected', lw=params_dict['lw'], color='m')
+    axis4.set_xlim(xlim_left, xlim_right)
+    axis4.xaxis.set_major_locator(MaxNLocator(integer=True))
+    y_ticks = axis4.get_ylim()
+    axis4.set_yticks([0, y_ticks[1] / 2, y_ticks[1]])
+    axis4.legend(fontsize=params_dict['fsize']//2)
+    axis4.set_ylabel(f"({params_dict['frequency_unit']})",
+                     fontsize=params_dict['fsize'])
+
+    print('plotting Statistical Significance ...')
+    axis5 = plt.subplot(6, 1, 5, sharex=axis1)
+    axis5.set_title('Statistical Significance')
+    joint_surprise_values = joint_surprise_dict['Js']
+    axis5.plot(center_of_analysis_window, joint_surprise_values,
+               lw=params_dict['lw'], color='k')
+    axis5.set_xlim(xlim_left, xlim_right)
+    axis5.set_ylim(params_dict['S_ylim'])
+    axis5.axhline(joint_surprise_significance, ls='-', color='r')
+    axis5.axhline(-joint_surprise_significance, ls='-', color='g')
+    axis5.text(t_winpos[30], joint_surprise_significance + 0.3, '$\\alpha +$',
+               color='r')
+    axis5.text(t_winpos[30], -joint_surprise_significance - 0.9, '$\\alpha -$',
+               color='g')
+    axis5.xaxis.set_major_locator(MaxNLocator(integer=True))
+    axis5.set_yticks([ue.jointJ(1-significance_level), ue.jointJ(0.5),
+                      ue.jointJ(significance_level)])
+    axis5.set_yticklabels([1-significance_level, 0.5, significance_level])
+
+    print('plotting Unitary Events ...')
+    axis6 = plt.subplot(6, 1, 6, sharex=axis1)
+    axis6.set_title('Unitary Events')
+    for n in range(n_neurons):
+        for trial, data_trial in enumerate(data):
+            spike_events_on_timescale = data_trial[n].magnitude
+            spike_events_on_trialscale = \
+                np.ones_like(data_trial[n].magnitude) * trial + \
```

Comment:
I see this thing for the 3rd time
hm
but you already spent a lot of time restructuring the logic. let it be as it is